src/main/java/com/acme/dbo/txlog/Facade.java

```diff
@@ -1,11 +1,147 @@
 package com.acme.dbo.txlog;
 
+import static java.lang.Math.abs;
+
 public class Facade {
-    public static void log(int message) {
-        System.out.println("primitive: " + message);
+    private static Integer integerAccumulator;
+    private static Byte byteAccumulator;
+    private static String stringAccumulator;
+    private static String PRIMITIVE_DECOR = "primitive: ";
+    private static String INTEGER_DECOR = PRIMITIVE_DECOR;
+    private static String BYTE_DECOR = PRIMITIVE_DECOR;
+    private static String CHAR_DECOR = "char: ";
+    private static String STRING_DECOR = "string: ";
+    private static String REFERENCE_DECOR = "reference: ";
+
+
+    public static void log(int message, boolean isDecorated) {
+        flushLastState(Facade.integerAccumulator, isDecorated, "Byte", "String");
```

Comment:
You can use static import to get rid of Facade.integerAccumulator to just integerAccumulator variable name.